components/brave_wallet/browser/simulation_response_parser_unittest.cc

```diff
@@ -1633,159 +1692,155 @@ TEST(SimulationResponseParserUnitTest, ParseSolanaWarnings) {
             "We suspect this transaction is malicious. Approving may lead to "
             "loss of funds.");
 
-  EXPECT_FALSE(simulation_response->simulation_results->error);
-  EXPECT_FALSE(
-      simulation_response->simulation_results->is_recent_blockhash_expired);
-  EXPECT_EQ(
-      simulation_response->simulation_results->expected_state_changes.size(),
-      0ULL);
+  EXPECT_FALSE(simulation_response->error);
+  EXPECT_EQ(simulation_response->expected_state_changes.size(), 0ULL);
 }
 
 TEST(SimulationResponseParserUnitTest, ParseSolanaNullableFields) {
   std::string json_fmt(R"(
     {
-      "status": "CHECKS_PASSED",
-      "action": "NONE",
-      "warnings": [],
-      "simulationResults": {
-        "isRecentBlockhashExpired": false,
-        "expectedStateChanges": [
-          {
-            "humanReadableDiff": "Send 2 USDT",
-            "suggestedColor": "DEBIT",
-            "rawInfo": {
-              "kind": "SPL_TRANSFER",
-              "data": {
-                "symbol": "USDT",
-                "name": "USDT",
-                "mint": "Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB",
-                "decimals": "6",
-                "supply": "1000000000",
-                "metaplexTokenStandard": "unknown",
-                "assetPrice":%s,
-                "diff": {
-                  "sign": "MINUS",
-                  "digits": "2000000"
+      "aggregated": {
+        "action": "NONE",
+        "warnings": [],
+        "expectedStateChanges": {
+          "8eekKfUAGSJbq3CdA2TmHb8tKuyzd5gtEas3MYAtXzrT": [
+            {
+              "humanReadableDiff": "Send 2 USDT",
+              "suggestedColor": "DEBIT",
+              "rawInfo": {
+                "kind": "SPL_TRANSFER",
+                "data": {
+                  "asset": {
+                    "symbol": "USDT",
+                    "name": "USDT",
+                    "mint": "Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB",
+                    "decimals": "6",
+                    "supply": "1000000000",
+                    "metaplexTokenStandard": "unknown",
+                    "price": null,
```

Comment:
There was a bug on this relating to this field not being substituted. This was recently fixed in a separate PR [1]. A review of these tests should follow to make sure they cover both fail and success routes to make sure we are in general testing what we think to be testing. An [2] issue has been created for this review.

[1] https://github.com/brave/brave-core/pull/25643
[2] https://github.com/brave/brave-browser/issues/41144